src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java

```diff
@@ -32,7 +32,6 @@ public FirebaseDataStorage() {
 
```

Comment:
L5: Don't use * imports, add all classes that you are using explicitly - it is much easier to follow then.
L21: PROJECT_ID, same for L22
L37: nit: field names should be constants as well
L45: nit: return document.exists() ? Optional.of(document) : Optional.empty();
L57: nit: I don't see much of use for this method, I think it can be inlined in the constructor.
L60: The file name should be a constant, see relevant comment in Bot for more details.
L69: nit: return FirestoreClient.getFirestore(); no need for the variable.

Reply:
Resolved: all except L57